packages/binding-coap/src/util.ts

```diff
@@ -0,0 +1,81 @@
+/********************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ * Document License (2015-05-13) which is available at
+ * https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+
+import { ProtocolHelpers } from "@node-wot/core";
+import { PropertyElement } from "wot-thing-description-types";
+
+const observeOpFilter = ["observeproperty", "unobserveproperty"];
+const readWriteOpFilter = ["readproperty", "writeproperty"];
+const eventOpFilter = ["subscribeevent", "unsubscribeevent"];
+
+function filterOpValues(opValues: string[], filterValues: string[]) {
+    return opValues.filter((opValue) => filterValues.includes(opValue));
+}
+
+/**
+ * Convenience function to filter out the `op` values "observeproperty" and
+ * "unobserveproperty" from a string array.
+ *
+ * @param opValues The `op` values to be filtered.
+ * @returns A filtered array that might be empty.
+ */
+export function filterPropertyObserveOperations(opValues: string[]) {
+    return filterOpValues(opValues, observeOpFilter);
+}
+
+/**
+ * Convenience function to filter out the `op` values "readproperty" and
+ * "writeproperty" from a string array.
+ *
+ * @param opValues The `op` values to be filtered.
+ * @returns A filtered array that might be empty.
+ */
+export function filterPropertyReadWriteOperations(opValues: string[]) {
+    return filterOpValues(opValues, readWriteOpFilter);
+}
+
+/**
+ * Convenience function to filter out the `op` values "subscribeevent" and
+ * "unsubscribeevent" from a string array.
+ *
+ * @param opValues The `op` values to be filtered.
+ * @returns A filtered array that might be empty.
+ */
+export function filterEventOperations(opValues: string[]) {
```

Comment:
Is this function used somewhere? I understand you added for completeness right? but given this is a private util file, maybe we can consider remove it...